contracts/VaultManager/VaultManager.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./IVaultManager.sol";
+import "./VaultManagerEvents.sol";
+import "../Vault/Vault.sol";
+import "poolz-helper-v2/contracts/GovManager.sol";
+
+contract VaultManager is IVaultManager, VaultManagerEvents,GovManager{
+    mapping(uint => address) public VaultIdToVault;
+    mapping(address => uint[]) public TokenToVaultIds;
```

Comment:
My initial idea was exactly that, to make a mapping of tokenAddress => vaultAddress. But this won't work because it is possible for a token to have multiple vaults, for example, the new PoolzBack can have multiple vaults of BUSD. Each of the BUSD vaults will be connected to the refund contract. This is why both LD and PoolzBack will have to store the vaultID.